### PR TITLE
Feature/app landing zone

### PR DESCRIPTION
**Pull request by QBY-ChristianHartmann**
# Description

As discussed in #(issue) we need to support xxx. Therefore we changed yyy.

# Change overview (tick true):

- [x] This introduces backward incompatible changes
- [ ] This adds a new backward compatible Feature
- [ ] This fixes a Bug

# Version information: 

- Previous Version: `major.minor.patch`

- Next Version based on [Semantic Versioning](https://semver.org/#summary) (see above): `major.minor.patch`

# How Has This Been Tested?

- [ ] Apply of all examples was successfull
- [ ] Test B

# Checklist:

- [ ] I have run tests and documented them above
- [x] I have performed a self-review of my own code
- [ ] I have updated the documentation

### CODE REVIEW

File: templates/main.tftpl
Comment (TheodorWulff) [inline templates/main.tftpl:1]: Let's at least add a comment here that states what this file should be used for in the future

File: templates/locals.tftpl [hunk 1]
```diff
@@ -0,0 +1,4 @@
+locals {
+  default_location = "${location}"
+  subsciption_logical_name = split("-", "${subscription_name}")[1]
```
Comment (TheodorWulff) [inline templates/locals.tftpl:3]: still needs ""

File: templates/main.tftpl
Comment (TheodorWulff) [inline templates/main.tftpl:1]: Let's at least add a comment here that states what this file should be used for in the future

File: templates/terraform.tftpl [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">=3.46.0"
+    }
+  }
+}
+provider "azurerm" {
+  features {}
+  skip_provider_registration = true
```
Comment (TheodorWulff) [inline templates/terraform.tftpl:11]: Make optional or remove

File: templates/virtual_machine.tftpl [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    version = "" #always check the current version and examples
+    virtual_machine_config = {
+        hostname = ""
+        location = local.location
+        admin_username = ""
+        size = ""
+        os_sku = ""
+    }
+    admin_password = ""
+    resource_group_name =
```
Comment (TheodorWulff) [inline templates/virtual_machine.tftpl:17]: I think we can add something here right?

File: terraform.tf [hunk 4]
```diff
@@ -12,12 +12,17 @@ terraform {
...
+      source = "salrashid123/http-full"
+      version = "1.3.1"
+    }
   }
 }
 
 provider "azurerm" {
   features {}
-  subscription_id = var.management_subscription_id
+  subscription_id = var.devops_subscription_id
```
Comment (TheodorWulff) [inline terraform.tf:24]: Verify which subscription actually has to be used

File: terraform.tfvars.sample [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+skip_provider_registration = #optional 
+vnet_config = { #optional 
+    dns_server = [""]
+    address_space = ""
+    subnets = {
+      databases = "",
+      application = ""
+    }
+}
+virtual_machine_template = #Optional
```
Comment (TheodorWulff) [inline terraform.tfvars.sample:24]: Optional is nice but what is expected here?

File: variables.tf [hunk 6]
```diff
@@ -41,4 +42,45 @@ variable "terraform_state_config" {
...
+  default = null
+}
+
+variable "skip_provider_registration" {
+  type        = bool
+  description = "Allows you to skip the provider registration when initilizing the azurerm provider. This is useful in development environments where not every provider can be registered."
+  default     = false
+}
+
+variable "virtual_machine_template" {
```
Comment (TheodorWulff) [inline variables.tf:82]: Instead of a boolean we could prefill the template already with values and leave it out if nothing was supplied

**Review by TheodorWulff — CHANGES_REQUESTED** [review]

**Review by ghost — COMMENTED** [review]
Please run formatter :)

File: terraform.tf (hunk 4 above)
Comment (TheodorWulff) [reply terraform.tf:24]: Nevermind

File: terraform.tfvars.sample (hunk 5 above)
Comment (TheodorWulff) [reply terraform.tfvars.sample:24]: And why is the variable then called virtual_machine_template?

File: variables.tf (hunk 6 above)
Comment (QBY-ChristianHartmann) [reply variables.tf]: ad discussed we leave the module like that

**Review by TheodorWulff — APPROVED** [review]

**Pull request merged**